Change name of libhdf5hl_fortran installed by autotools

lrknox opened the pull request:
to libhdf5_hl_fortran to be consistent with CMake install and with other hl lib files.  Switched corresponding symlink to libhdf5hl_fortran.

Fixes issue #4684.

lrknox committed:
Change name of libhdf5hl_fortran installed by autotools to libhdf5_hl_fortran to be
consistent with CMake install and with other hl lib files.  Switched
corresponding symlink to libhdf5hl_fortran.
Fixes issue #4684.

lrknox committed:
Update h5fc.in to use renamed libhdf5_hl_fortran.
Fix typo in code to create link to libhdf5_hl_fortran.a.

mattjala commented:
The issue was flagged for 1.16 - is this the kind of thing we should  delay until the next major update to avoid surprising users?

derobins commented:
This can go in now, but should be develop only. Renaming libraries should only be done in a major release.

derobins commented:
Also, this needs a RELEASE.txt entry

derobins commented:
Decision is to:

1) Remove the symlink
2) Rename the library to be consistent with CMake and other hl libraries
3) This will go to develop only
4) Will announce in a RELEASE.txt note and on the forum

lrknox committed:
Removed code to create symlink for previous name of renamed lib
libhdf5hl_fortran.
Add RELEASE.txt entry about changeing name libhdf5hl_fortran to
libhdf5_hl_fortran.
